### PR TITLE
users can't edit inactive services

### PR DESCRIPTION
**Pull request by leohemsted**
if a service gets archived, then we already block them from using any unrevoked api keys (https://github.com/alphagov/notifications-api/blob/master/app/authentication/auth.py#L66), but previously didn't stop them creating new templates, uploading CSVs, etc.

Now, block any calls to the API if the service is inactive and the user's not a platform admin

### CODE REVIEW

File: tests/app/notify_client/test_notify_admin_api_client.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    platform_admin_user(str(uuid.uuid4()))
+], ids=['api_user', 'platform_admin'])
+@pytest.mark.parametrize('service', [
+    service_json(active=True),
+    None
+], ids=['active_service', 'no_service'])
+def test_active_service_can_be_modified(app_, method, user, service):
+    api_client = NotifyAdminAPIClient('api_key', 'base_url', 'service_id')
+
+    with app_.test_request_context() as request_context, app_.test_client() as client:
```
Comment (imdadahad) [inline tests/app/notify_client/test_notify_admin_api_client.py]: Can you use the client in conftest here?

Comment (leohemsted) [inline tests/app/notify_client/test_notify_admin_api_client.py]: unfortunately not, since I need to modify the request_context object :(

Comment (leohemsted) [inline tests/app/notify_client/test_notify_admin_api_client.py]: unless in conftest i attach request_context to client as a variable or similar, which i'm loathe to do cos it'll be undocumented

Comment (imdadahad) [inline tests/app/notify_client/test_notify_admin_api_client.py]: fair enough, looks good 👍

**Pull request merged**